src/jdk.management/share/classes/jdk/management/HotSpotAOTCacheMXBean.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, Microsoft, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.management;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.PlatformManagedObject;
+import java.util.concurrent.ForkJoinPool;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+/**
+ * Management interface for the JDK's Ahead of Time (AOT) optimizations.
+ *
+ * Currently, {@code HotSpotAOTCacheMXBean} defines one operation at this time to end the AOT recording.
+ * More operations and/or properties may be added in a future release.
+ *
+ * <p> The management interface is registered with the platform {@link MBeanServer
+ * MBeanServer}. The {@link ObjectName ObjectName} that uniquely identifies the management
+ * interface within the {@code MBeanServer} is: "jdk.management:type=HotSpotAOTCache".
+ *
+ * <p> Direct access to the MXBean interface can be obtained with
+ * {@link ManagementFactory#getPlatformMXBean(Class)}.
+ *
+ * @since 26
+ */
+public interface HotSpotAOTCacheMXBean extends PlatformManagedObject {
+       /**
+       * If an AOT recording is in progress, ends the recording. This operation completes
+       * after the AOT artifacts have been completely written.
+       *
+       * <p>The JVM will start recording AOT artifacts upon start-up if certain JVM options are
+       *  given in the command-line. The recording will stop when the JVM exits, or when
+       * the {@code endRecording} method is called. Examples:
+       *
+       * <p> java -XX:AOTCacheOutput=app.aot ....
+       *
+       * <blockquote>
+       *       The JVM will record optimization information about the current application
+       *       into the AOT cache file app.aot. In a future execution of this application,
```

Comment:
We don't strictly generate the info into `app.aot` - we record it and then do an assembly phase (either implicitly with AOTCacheOutput or explicitly with AOTMode=) to generate the cache:
```suggestion
       *       The JVM will record optimization information about the current application
       *       that will be used to generate the AOT cache file app.aot. In a future execution of this application,
```